deployment/migration-assistant-solution/lib/solutions-stack.ts

```diff
@@ -0,0 +1,179 @@
+        super(scope, id, props);
+        this.templateOptions.templateFormatVersion = '2010-09-09';
+        new CfnMapping(this, 'Solution', {
+            mapping: {
+                Config: {
+                    CodeVersion: props.solutionVersion,
+                    KeyPrefix: `${props.solutionName}/${props.solutionVersion}`,
+                    S3Bucket: props.codeBucket,
+                    SendAnonymousUsage: 'No',
+                    SolutionId: props.solutionId
+                }
+            },
+            lazy: false,
+        });
+
+        const stageParameter = new CfnParameter(this, 'Stage', {
+            type: 'String',
+            description: 'Specify the stage identifier which will be used in naming resources, e.g. dev,gamma,wave1',
+            default: 'dev',
+            noEcho: false
+        });
+
+        const appRegistryAppARN = applyAppRegistry(this, stageParameter.valueAsString, props)
+        const vpc = new Vpc(this, 'Vpc', {});
+
+        new CfnDocument(this, "BootstrapShellDoc", {
+            name: `SSM-${stageParameter.valueAsString}-BootstrapShell`,
+            documentType: "Session",
+            content: {
+                "schemaVersion": "1.0",
+                "description": "Document to hold regional settings for Session Manager",
+                "sessionType": "Standard_Stream",
+                "inputs": {
+                    "cloudWatchLogGroupName": "",
+                    "cloudWatchEncryptionEnabled": true,
+                    "cloudWatchStreamingEnabled": false,
+                    "kmsKeyId": "",
+                    "runAsEnabled": false,
+                    "runAsDefaultUser": "",
+                    "idleSessionTimeout": "60",
+                    "maxSessionDuration": "",
+                    "shellProfile": {
+                        "linux": "cd /opensearch-migrations && sudo -s"
+                    }
+                }
+            }
+        })
+
+        const solutionsUserAgent = `AwsSolution/${props.solutionId}/${props.solutionVersion}`
+        const cfnInitConfig: InitElement[] = [
+            InitCommand.shellCommand(`echo "export MIGRATIONS_APP_REGISTRY_ARN=${appRegistryAppARN}; export MIGRATIONS_USER_AGENT=${solutionsUserAgent}" > /etc/profile.d/solutionsEnv.sh`),
+            InitFile.fromFileInline("/opensearch-migrations/initBootstrap.sh", './initBootstrap.sh', {
+                mode: "000744"
+            }),
+        ]
+
+        const bootstrapRole = new Role(this, 'BootstrapRole', {
+            assumedBy: new ServicePrincipal('ec2.amazonaws.com'),
+            description: 'EC2 Bootstrap Role'
+        });
+        bootstrapRole.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AdministratorAccess'))
+
+        new InstanceProfile(this, 'BootstrapInstanceProfile', {
+            instanceProfileName: `bootstrap-${stageParameter.valueAsString}-instance-profile`,
```

Comment:
To resolve this bug: https://opensearch.atlassian.net/browse/MIGRATIONS-2027, can you also include region in this name

Reply:
Done!